applications/googlehome-meets-dotnetcontainers/GoogleHomeAspNetCoreDemoServer/Dialogflow/Intents/Vision/VisionBacktoallHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright(c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GoogleHomeAspNetCoreDemoServer.Dialogflow.Intents.Vision
+{
+    /// <summary>
+    /// Handler for "vision.backtoall" DialogFlow intent.
+    /// </summary>
+    [Intent("vision.backtoall")]
+    public class VisitionBacktoallHandler : BaseVisionHandler
+    {
+        /// <summary>
+        /// Initializes class with the conversation.
+        /// </summary>
+        /// <param name="conversation">Conversation</param>
+        public VisitionBacktoallHandler(Conversation conversation) : base(conversation)
+        {
+        }
+
+        /// <summary>
+        /// Handle the intent.
+        /// </summary>
+        /// <param name="req"></param>
+        /// <returns></returns>
+        public override async Task<object> Handle(ConvRequest req)
+        {
+            // Unfocus the image
+            _conversation.State.FocusedImage = null;
+
+            // Update state with list of pictures
+            var images = _conversation.State.ImageList;
+            var imagesList = images.Select(x => $"<li><img src=\"{x.Url}\" alt=\"{x.Title}\" style=\"width:200px\" /></li>");
```

Comment:
What if the url or title contains a quote character?

Reply:
These are image urls returned from Google Custom Search, so it won't contain quote character?

Reply:
[RFC 3986](https://tools.ietf.org/html/rfc3986#section-2.3) specified that quote is not an unreserved character, so they should always be encoded in a URL.
However, the title could contain a quote (or any other) character.

Reply:
So what do we do here? Do we need to use some encoder to make sure the title is safe?

Reply:
I looked at using a razor page, but trying to use razor outside of MVC is a total hassle.

How about using https://docs.microsoft.com/en-us/dotnet/api/system.net.webutility.htmlencode?view=netcore-2.0#System_Net_WebUtility_HtmlEncode_System_String_?

Reply:
This is the final remaining thing I'd like to see fixed before merging.

Reply:
Yes, me too :-) I plan to fix the rest of the comments today, so we can merge it later today or tomorrow.

Reply:
Fixed. All the titles are encoded now.

Reply:
All the titles are encoded now